src/main/java/org/jabref/gui/FXDialogService.java

```diff
@@ -300,4 +303,13 @@ private FileChooser getConfiguredFileChooser(FileDialogConfiguration fileDialogC
     public boolean showPrintDialog(PrinterJob job) {
         return job.showPrintDialog(mainWindow);
     }
+
+    @Override
+    public Optional<String> showFileOpenFromArchiveDialog(Path archivePath) throws IOException {
+        try (ZipFile zipFile = new ZipFile(archivePath.toFile(), ZipFile.OPEN_READ)) {
```

Comment:
You could take the chance and implement the zip/jar reading using Paths class:
https://docs.oracle.com/javase/7/docs/technotes/guides/io/fsp/zipfilesystemprovider.html